src/common/mfu_flist_copy.c

```diff
@@ -92,67 +100,78 @@ static mfu_copy_stats_t mfu_copy_stats;
 static mfu_copy_file_cache_t mfu_copy_src_cache;
 static mfu_copy_file_cache_t mfu_copy_dst_cache;
 
-static int mfu_copy_open_file(const char* file, int read_flag,
-        mfu_copy_file_cache_t* cache, mfu_copy_opts_t* mfu_copy_opts)
+static void mfu_copy_open_file(const char* file, int read_flag,
+        mfu_copy_file_cache_t* cache, mfu_copy_opts_t* mfu_copy_opts,
+        mfu_file_t* mfu_file)
 {
-    int newfd = -1;
-
     /* see if we have a cached file descriptor */
     char* name = cache->name;
     if (name != NULL) {
         /* we have a cached file descriptor */
-        int fd = cache->fd;
+        int fd         = cache->fd;
+#ifdef DAOS_SUPPORT
+        dfs_obj_t* obj = cache->obj;
+#endif
         if (strcmp(name, file) == 0 && cache->read == read_flag) {
             /* the file we're trying to open matches name and read/write mode,
              * so just return the cached descriptor */
-            return fd;
+            if (fd > 0) {
```

Comment:
I think an fd==0 can be a valid open file descriptor, as well.
```suggestion
            if (fd >= 0) {
```
We should double check that it get initialized correctly.

Reply:
@adammoody The fd value is initialized to -1 when the new mfu_file is created:

```C
/* return a newly allocated mfu_file structure, set default values on its fields */
mfu_file_t* mfu_file_new(void)
{
    mfu_file_t* mfile = (mfu_file_t*) MFU_MALLOC(sizeof(mfu_file_t));
    mfile->type       = POSIX;
    mfile->fd         = -1;
    mfile->only_daos  = false;
#ifdef DAOS_SUPPORT
    mfile->obj        = NULL;
    mfile->dfs        = NULL;
#endif
    return mfile;
}
```